mmdet/core/hook/ema.py

```diff
@@ -0,0 +1,121 @@
+from mmcv.runner.hooks import HOOKS, Hook
+from mmcv.parallel import MMDataParallel, MMDistributedDataParallel
+import torch.nn as nn
+import math
+
+
+def is_parallel(model):
+    return type(model) in (
+        nn.parallel.DataParallel, nn.parallel.DistributedDataParallel,
+        MMDataParallel, MMDistributedDataParallel)
+
+
+class BaseEMAHook(Hook):
+    r"""Exponential Moving Average Hook.
+
+    Use Exponential Moving Average on all parameters of model in training
+    process. All parameters have a ema backup, which update by the formula
+    as below. EMAHook takes priority over EvalHook and CheckpointHook. Note,
+    the original model parameters are actually saved in ema field after train.
+
+    Args:
+        skip_bn_running_stats (bool): Whether to skip the batchnorm running stats
+            (running_mean, running_var), it does not perform the ema operation.
+            Default to False.
+        interval (int): Update ema parameter every interval iteration.
+            Defaults to 1.
+        resume_from (str, Optional): The checkpoint path. Defaults to None.
+    """
+
+    def __init__(self, skip_bn_running_stats=False, interval=1, resume_from=None):
+        self.skip_bn_running_stats = skip_bn_running_stats
```

Comment:
skip_bn_running_stats -> skip_buffer, as it essentially also affects other running buffers more than BN running status.